private/rules/java_export.bzl

```diff
@@ -291,6 +298,11 @@ def maven_export(
         for target, targetExclusions in exclusions.items()
     }
 
+    exclusion_source_json_strings = [
+        _json.write_artifact_spec(artifact)
+        for artifact in parse.parse_artifact_spec_list(exclusion_source)
+    ]
```

Comment:
I understand the need for the separate field on java_export, but I wonder if it would make sense to merge the two and just passing a single `exclusions` dict to `pom_file`?

At a high level:
parse and filter the `exclusion_source` list so it only includes dependencies with exclusions
convert the list so it is a dict in the same form as `exclusions`
merge the two dicts and pass to `pom_file`

Reply:
Good question. It's not straightforward:
* `pom_file`'s `exclusion` has labels for keys, not strings. I think that makes it safer to use so I'm not sure we should change that. If we tried moving `the computation of the key` outside of `pom_file` it would have to be a label, and we'd get back to needed to encode in snake case like we to do, and guess the repo (`@maven`).
* if we we move to stings as keys, and move the computation of the key` outside of `pom_file`, we wouldn't have access to the list of dependencies `expanded_maven_deps` that we use to filter the exclusions, so we'd likely forego the filtering and then `pom_file` would need to ignore irrelevant dependencies (again a bit more error prone for explicitly added exclusions).